pkgs/by-name/pz/pzip/package.nix

```
@@ -0,0 +1,34 @@
+{ lib
+, buildGo121Module
```

Maybe you want to follow the "[Changing implicit attribute defaults](https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/README.md#changing-implicit-attribute-defaults)" guide and replace `buildGo121Module` with the generic `buildGoModule` and add the following part to the `pkgs/top-level/all-packages.nix` file:
```nix
  pzip = callPackage ../by-name/pz/pzip/package.nix {
    buildGoModule = buildGo121Module;
  };
```

```suggestion
, buildGoModule
```

---

I don't think maintainers will recall this override in `all-packages.nix` if their packages are in `pkgs/by-name`, so making `buildGo121Module` explicit seem preferable to me. It's just a matter of taste, anyway.

---

> It's just a matter of taste, anyway.

Definitely, as the guide is also only a reflection of the taste of some individual resp. a consensus of a group of individuals. Personally I don't care, i was recently just happy that there was a document stating _how_ something should be done, as otherwise there are lots of different "styles" in the nixpkgs.

Thus, please feel free to either ignore my comment or adjust the code to guidelines.

---

> i was recently just happy that there was a document stating how something should be done

I really hope that part is only about migrating existing packages, but it's not. I don't see forcing generic attributes in new packages as a consequence of RFC 140, I'd like to ask @infinisil for clarification.

> as otherwise there are lots of different "styles" in the nixpkgs.

The use of `buildGo121Module` is indeed [already very divided](https://github.com/search?q=repo%3ANixOS%2Fnixpkgs%20buildGo121Module&type=code) in nixpkgs.

> Thus, please feel free to either ignore my comment or adjust the code to guidelines.

I see your point, and I'm not approving this PR for an unresolved conservation.

---

> I really hope that part is only about migrating existing packages, but it's not. I don't see forcing generic attributes in new packages as a consequence of RFC 140, I'd like to ask @infinisil for clarification.

It's not related to RFC 140 directly. I guess that part of the documentation would be better suited in `pkgs/README.md` instead of `pkgs/by-name/README.md` for now.

But in any case, I definitely think using generic attributes like `buildGoModule` is better than `buildGo121Module`, exactly because of the reason written down in [the linked section](https://github.com/NixOS/nixpkgs/blob/f99e5f03cc0aa231ab5950a15ed02afec45ed51a/pkgs/by-name/README.md#changing-implicit-attribute-defaults): It allows keeping the `.override` interface more stable.

For now it's unfortunate that adjusting these arguments requires editing `all-packages.nix` still, which will inevitably push people to not adding them, so in a way RFC 140 contributes to decreasing the stability of `.override`. I'd like to improve this situation with future efforts.

Still, I value user interface stability more over developer experience, so using generic attributes would be my preference.